.github/workflows/install-valkey/action.yml

```diff
@@ -0,0 +1,97 @@
+name: Install Valkey
+
+inputs:
+    engine-version:
+        description: "Engine version to install"
+        required: true
+        type: string
+    target:
+        description: "Specified target toolchain, ex. x86_64-unknown-linux-gnu"
+        type: string
+        required: true
+        options:
+            - x86_64-unknown-linux-gnu
+            - aarch64-unknown-linux-gnu
+            - x86_64-apple-darwin
+            - aarch64-apple-darwin
+            - aarch64-unknown-linux-musl
+            - x86_64-unknown-linux-musl
+
+env:
+    CARGO_TERM_COLOR: always
+    VALKEY_MIN_VERSION: "7.2.5"
+
+runs:
+    using: "composite"
+
+    steps:
+        - name: Cache Valkey
+           # TODO: remove the musl ARM64 limitation when https://github.com/actions/runner/issues/801 is resolved
+          if: ${{ inputs.target != 'aarch64-unknown-linux-musl' }}
+          uses: actions/cache@v4
+          id: cache-valkey
+          with:
+            path: |
+                ~/valkey
+            key: valkey-${{ inputs.engine-version }}-${{ inputs.target }}
+
+        - name: Build Valkey
+          if: ${{ steps.cache-valkey.outputs.cache-hit != 'true' }}
+          shell: bash
+          run: | 
+            echo "Building valkey ${{ inputs.engine-version }}"
+            cd ~
+            git clone https://github.com/valkey-io/valkey.git
+            cd valkey
+            git checkout ${{ inputs.engine-version }}
+            make BUILD_TLS=yes
+
+        - name: Install Valkey
+          shell: bash
+          run: |
+            cd ~/valkey
+            if command -v sudo &> /dev/null
+            then
+                echo "sudo command exists"
+                sudo make install
+            else
+                echo "sudo command does not exist"
+                make install
+            fi
+            echo 'export PATH=/usr/local/bin:$PATH' >>~/.bash_profile
+
+        - name: Check if the current version is a Valkey release
+          id: is_valkey_release
+          uses: ./.github/workflows/compare-versions
+          with:
+              version1: ${{ inputs.engine-version }}
+              version2: ${{ env.VALKEY_MIN_VERSION }}
```

Comment:
You dont need this input, it is always the same number, use it as an env-param in the compare action

Reply:
The `compare versions` action shouldn't be coupled with this specific check, it's a general version comparison action so we can use it later on for other checks

Reply:
It is a bit YAGNI.
You are making the CI more complicated when currently it is not needed and it is hard to see why it will be. If the point that it's needed will come, it can be added then. The CI is already a rabbit hole.

Reply:
I disagree. 
Version comparison is a utility commonly implemented in various programming languages as an importable module. It's a convention to store version comparison logic within a function or utility library. Since we don't have a ready-to-use github action for that which works on Alpine, I created one. I don't see any advantage in hardcoding Valkey's minimum version to the version comparison function..